server/src/sylvester/event.py

```diff
@@ -0,0 +1,193 @@
+
+from abc import ABC
+from dataclasses import dataclass
+
+"""
+This class defines ALL events generated by the Sylvester v1 contract.
+
+Event List: Lend, Rent, StopRent, StopLend, RentClaimed
+"""
+
+@dataclass
+class Event(ABC):
+    tx_hash: str
+    tx_offset: int
+
+    @staticmethod
+    def parse_id(id):
+        """
+        Parses '_id' to tx_hash and tx_offset. 
+
+        Args:
+            id (str): mongodb '_id' e.i. 0xf79edc5500218427d22d4215799fb51064c633ca919c0406b830ed5dc3c6eb1a_100
+
+        Returns: 
+            str: tx_hash
+            int: tx_offset
+        """
+        id = id.split('_')
+
+        return id[0], int(id[1])
+
+
+@dataclass
+class LendEvent(Event):
+    is721: bool
+    lenderAddress: str
+    nftAddress: str
+    tokenID: str
+    lendingID: int
+    maxRentDuration: int
+    dailyRentPrice: float
+    lendAmount: int
+    paymentToken: int
+
+
+    @classmethod
+    def from_doc(cls, doc):
+        """
+        Parses a mongodb document into a 'Lend' Sylvester v1 event.
+
+        Args:
+            doc (Dict): a mongodb document
+
+        Returns:
+            _type_: instance of this class with the correct
```

Comment:
if there is no type definitions, let's write a todo comment to not forget to add it later on